src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -82,6 +74,25 @@ internal class RoslynPackage : AbstractPackage
             _solutionEventMonitor = new SolutionEventMonitor(_workspace);
         }
 
+        private async Task InitializeFromBackgroundAsync(CancellationToken cancellationToken)
```

Comment:
❗️ Asynchronous methods should not require the caller handle thread affinitization. Add the following on the first line to ensure the body of the method executes on a background thread:

```csharp
await TaskScheduler.Default;
```

Reply:
so. you are saying each async method should switch to whatever thread they want before running? is this new pattern? Roslyn didn't use that pattern before. @dotnet/roslyn-ide  @dotnet/roslyn-analysis 

so basically caller/callee should turn to whatever thread we want to run is what you are saying.

Reply:
this requires JTF to invade all over Roslyn code. so are we okay with that?

Reply:
Indeed, Roslyn has always tried to be explicit about what is being called where....

Reply:
We consider it a Good Thing that an async method hides the threads it needs to run as an implementation detail. It makes it far easier/safer for many callers of a method to simply invoke the async method. The async method can then acquire a need for the main thread later on and no callers have to be updated. So ya, per the first threading rule, async methods should switch to and from the main thread for themselves rather than the async code they're about to call.

Switching to a threadpool thread doesn't require JTF. But switching back to the main thread *should* be done with JTF to avoid deadlocks.